src/mimir_coordinator.py

```diff
@@ -37,37 +42,39 @@
     MimirRole.compactor: 1,
     MimirRole.distributor: 1,
 }
-"""The set of roles that need to be allocated for the 
+"""The set of roles that need to be allocated for the
 deployment to be considered robust according to the official recommendations/guidelines."""
 
 
 class MimirCoordinator:
+    """Mimir coordinator."""
+
     def __init__(self, relations: List[Relation]):
         self.relations = relations
 
     def is_coherent(self):
         """Return True if the roles list makes up a coherent mimir deployment."""
         roles = self.roles()
-        return set(roles).issubset(MINIMAL_DEPLOYMENT)
+        return roles and set(roles).issubset(MINIMAL_DEPLOYMENT)
```

Comment:
[praise] heh, good point.

Reply:
actually, my bad, we're doing this all wrong. It should be issuperset!

Reply:
or `MINIMAL_DEPLOYMENT.issubset(roles)` and then we don't need the `roles and` anymore.